wandb/integration/yolov8/yolov8.py

```diff
@@ -0,0 +1,250 @@
+                }
+            )
+
+    def on_train_end(self, trainer: BaseTrainer):
+        """On train end we log all the media, including plots, images and best model artifact to Weights & Biases."""
+        if not isinstance(trainer, ClassificationTrainer):
+            self.run.log(
+                {
+                    "plots": [
+                        wandb.Image(str(image_path), caption=image_path.stem)
+                        for image_path in trainer.save_dir.glob("*.png")
+                    ],
+                    "val_images": [
+                        wandb.Image(str(image_path), caption=image_path.stem)
+                        for image_path in trainer.validator.save_dir.glob("val*.jpg")
+                    ],
+                },
+            )
+
+        if trainer.best.exists():
+            self.run.log_artifact(
+                str(trainer.best),
+                type="model",
+                name=f"{self.run.name}_{trainer.args.task}.pt",
+                aliases=["best", f"epoch_{trainer.epoch + 1}"],
+            )
+
+    def on_model_save(self, trainer: BaseTrainer):
+        """On model save we log the model as an artifact to Weights & Biases."""
+        self.run.log_artifact(
+            str(trainer.last),
+            type="model",
+            name=f"{self.run.name}_{trainer.args.task}.pt",
+            aliases=["last", f"epoch_{trainer.epoch + 1}"],
+        )
+
+    def teardown(self, _trainer: BaseTrainer):
+        """On teardown we finish the Weights & Biases run and set it to None."""
+        self.run.finish()
+        self.run = None
+
+    @property
+    def callbacks(
+        self,
+    ):
+        """Contains all the relevant callbacks to add to the YOLO model for the Weights & Biases logging."""
+        return {
+            "on_pretrain_routine_start": self.on_pretrain_routine_start,
+            "on_pretrain_routine_end": self.on_pretrain_routine_end,
+            "on_train_epoch_start": self.on_train_epoch_start,
+            "on_train_epoch_end": self.on_train_epoch_end,
+            "on_fit_epoch_end": self.on_fit_epoch_end,
+            "on_train_end": self.on_train_end,
+            "on_model_save": self.on_model_save,
+            "teardown": self.teardown,
+        }
+
+
+def add_callbacks(
+    yolo: YOLO,
+    run_name: Optional[str] = None,
+    project: Optional[str] = None,
+    tags: Optional[List[str]] = None,
+    resume: Optional[str] = None,
+    **kwargs,
+) -> YOLO:
+    """A YOLO model wrapper that tracks metrics, and logs models to Weights & Biases.
+
+    Args:
+        yolo: A YOLO inherited from `:class:ultralytics.yolo.engine.model.YOLO` to add the callbacks to.
+        run_name: The name of the Weights & Biases run.
+        project: The name of the Weights & Biases project.
+        tags: A list of tags to add to the Weights & Biases run.
+        resume: Whether to resume the Weights & Biases run if it exists.
```

Comment:
see my comments from before. If this is the function we expose to the users. I think it would be useful in the documentation to describe all the defaults we chose for them

Reply:
Updated the docstrings to provide the information about defaults.

Reply:
Awesome! Thanks